cloud/aws-cloudwatch-metric-stream/aws_cloudwatch_metric_stream.tf

```diff
@@ -0,0 +1,59 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+resource "aws_s3_bucket" "sfx_metric_stream" {
+  bucket = "splunk-metric-streams-s3-${data.aws_caller_identity.current.account_id}-${data.aws_region.current.name}"
+}
+
+resource "aws_cloudwatch_log_group" "sfx_metric_stream" {
+  name              = "/aws/kinesisfirehose/splunk-metric-streams-${data.aws_region.current.name}"
+  retention_in_days = 14
+}
+
+resource "aws_cloudwatch_log_stream" "http_log_stream" {
+  name           = "HttpEndpointDelivery"
+  log_group_name = aws_cloudwatch_log_group.sfx_metric_stream.name
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "sfx_metric_stream" {
+  name        = "splunk-metric-streams-${data.aws_region.current.name}"
+  destination = "http_endpoint"
+
+  s3_configuration {
+    role_arn           = aws_iam_role.sfx_firehose_role.arn
+    bucket_arn         = aws_s3_bucket.sfx_metric_stream.arn
+    buffer_size        = 1
+    buffer_interval    = 60
+    compression_format = "GZIP"
+
+    cloudwatch_logging_options {
+      enabled         = true
+      log_group_name  = aws_cloudwatch_log_group.sfx_metric_stream.name
+      log_stream_name = aws_cloudwatch_log_stream.http_log_stream.name
+    }
+  }
+
+  http_endpoint_configuration {
+    url                = "${var.sfx_ingest_url}/v1/cloudwatch_metric_stream"
+    name               = "SignalFx"
+    access_key         = var.sfx_access_token
+    buffering_size     = 1
+    buffering_interval = 60
+    role_arn           = aws_iam_role.sfx_firehose_role.arn
+    s3_backup_mode     = "FailedDataOnly"
+
+    cloudwatch_logging_options {
+      enabled         = true
+      log_group_name  = aws_cloudwatch_log_group.sfx_metric_stream.name
+      log_stream_name = aws_cloudwatch_log_stream.http_log_stream.name
+    }
+
+    request_configuration {
+      content_encoding = "GZIP"
+    }
+  }
+
+  tags = {
+    "splunk-metric-streams-firehose" = "${data.aws_region.current.name}"
```

Comment:
I am not sure if there is a real convention on aws tag naming but it seems different than the usual camlcase style

Reply:
After further checking it seems it comes from Splunk official cloudformation template https://o11y-public.s3.amazonaws.com/aws-cloudformation-templates/release/template_metric_streams_regional.yaml so it is better to keep it as it is.